protobuf-codegen/src/message.rs

```diff
@@ -379,10 +384,17 @@ impl<'a> MessageGen<'a> {
                                 FieldKind::Oneof(..) => unreachable!(),
                             }
                         };
+
+                        let field_type = &field.full_storage_type().to_string();
+                        if self.serde_derive_enabled() && field_type.contains("::protobuf::SingularPtrField<") {
```

Comment:
This should be a match of field kind, not on string, e. g.

```
                    if let FieldKind::Singular(singular) = field.kind {
                        if let SingularFieldFlag::WithFlag { option_kind, .. } = singular.flag {
                            if option_kind == OptionKind::SingularField {
                                // write custom serializer
                            }
                        }
                    }
```

Reply:
Thanks for all of this instruction!